app/controllers/predicates/AuthActionHelper.scala

```diff
@@ -35,11 +35,15 @@ trait AuthActionHelper extends BetaChecks {
   def authorisedWithData(ern: String): ActionBuilder[DataRequest, AnyContent] =
     authorised(ern) andThen getData()
 
-  private def authorisedWithBetaGuardData(ern: String, betaGuard: (String, Result)): ActionBuilder[DataRequest, AnyContent] =
+  def authorisedWithBetaGuardData(ern: String, betaGuard: (String, Result)): ActionBuilder[DataRequest, AnyContent] =
     authorised(ern) andThen betaAllowList(betaGuard) andThen getData()
 
-  def authorisedDataRequest(ern: String)(block: DataRequest[_] => Result): Action[AnyContent] =
-    authorisedWithData(ern)(block)
+  def authorisedDataRequest(ern: String, betaGuard: Option[(String, Result)] = None)(block: DataRequest[_] => Result): Action[AnyContent] =
+    if (betaGuard.isEmpty) {
+      authorisedWithData(ern)(block)
+    } else {
+      authorisedWithBetaGuardData(ern, betaGuard.get)(block)
+    }
```

Comment:
Using a fold may be better:
```suggestion
    betaGuard.fold(authorisedWithData(ern)(block))(authorisedWithBetaGuardData(ern, _)(block))
```

Reply:
And may be worth applying this suggestion to the `authorisedDataRequestAsync` method

Reply:
Updated at https://github.com/hmrc/emcs-tfe-frontend/pull/130/commits/97479a74f01fb8797a946d669e6dd72e6470727a